src/ResourceManager/Compute/Commands.Compute/Generated/VirtualMachineScaleSet/VirtualMachineScaleSetUpdateMethod.cs

```diff
@@ -110,4 +113,877 @@ protected PSArgument[] CreateVirtualMachineScaleSetUpdateParameters()
+        public string ImageReferenceId { get; set; }
+
+        [Parameter(
+            ParameterSetName = "DefaultParameter",
+            Mandatory = false)]
+        [ValidateNotNullOrEmpty]
+        public bool? DisablePasswordAuthentication { get; set; }
+
+        [Parameter(
+            ParameterSetName = "DefaultParameter",
+            Mandatory = false)]
+        [ValidateNotNullOrEmpty]
+        public Hashtable Tag { get; set; }
+
+        [Parameter(
+            ParameterSetName = "DefaultParameter",
+            Mandatory = false)]
+        [ValidateNotNullOrEmpty]
+        public string PlanName { get; set; }
+
+        [Parameter(
+            ParameterSetName = "DefaultParameter",
+            Mandatory = false)]
+        [ValidateNotNullOrEmpty]
+        public int? MaxUnhealthyUpgradedInstancePercent { get; set; }
+
+        [Parameter(
+            ParameterSetName = "DefaultParameter",
+            Mandatory = false)]
+        [ValidateNotNullOrEmpty]
+        public string ImageReferencePublisher { get; set; }
+
+        [Parameter(
+            ParameterSetName = "DefaultParameter",
+            Mandatory = false)]
+        [ValidateNotNullOrEmpty]
+        public string PlanProduct { get; set; }
+
+        [Parameter(
+            ParameterSetName = "DefaultParameter",
+            Mandatory = false)]
+        [ValidateNotNullOrEmpty]
+        public string ImageUri { get; set; }
+
+        [Parameter(
+            ParameterSetName = "DefaultParameter",
+            Mandatory = false)]
+        [ValidateNotNullOrEmpty]
+        public string SkuTier { get; set; }
+
+        [Parameter(
+            ParameterSetName = "DefaultParameter",
+            Mandatory = false)]
+        [ValidateNotNullOrEmpty]
+        public string[] VhdContainer { get; set; }
+
+        [Parameter(
+            ParameterSetName = "DefaultParameter",
+            Mandatory = false)]
+        [ValidateNotNullOrEmpty]
+        public string LicenseType { get; set; }
+
+        [Parameter(
+            ParameterSetName = "DefaultParameter",
+            Mandatory = false)]
+        [ValidateNotNullOrEmpty]
+        public ResourceIdentityType? IdentityType { get; set; }
+
+        [Parameter(
+            ParameterSetName = "DefaultParameter",
+            Mandatory = false)]
+        [ValidateNotNullOrEmpty]
+        public string SkuName { get; set; }
+
+        [Parameter(
+            ParameterSetName = "DefaultParameter",
+            Mandatory = false)]
+        [ValidateNotNullOrEmpty]
+        public string PlanPromotionCode { get; set; }
+
+        [Parameter(
+            ParameterSetName = "DefaultParameter",
+            Mandatory = false)]
+        [ValidateNotNullOrEmpty]
+        public int? MaxUnhealthyInstancePercent { get; set; }
+
+        [Parameter(
+            ParameterSetName = "DefaultParameter",
+            Mandatory = false)]
+        [ValidateNotNullOrEmpty]
+        public int? SkuCapacity { get; set; }
```

Comment:
Is this the parameter that is used for scaling?  What is the reason for using a nullable type with the 'ValidateNotNullOrEmpty' attribute..  We should just use the non-nullable type, if this value cannot be null.